react/features/prejoin/functions.js

```diff
@@ -0,0 +1,194 @@
+}
+
+/**
+ * Returns the text for the prejoin status bar.
+ *
+ * @param {Object} state - The state of the app.
+ * @returns {string}
+ */
+export function getDeviceStatusText(state: Object): string {
+    return state['features/prejoin'].deviceStatusText;
+}
+
+/**
+ * Returns the type of the prejoin status bar: 'ok'|'warning'.
+ *
+ * @param {Object} state - The state of the app.
+ * @returns {string}
+ */
+export function getDeviceStatusType(state: Object): string {
+    return state['features/prejoin'].deviceStatusType;
+}
+
+/**
+ * Selector for getting the prejoin video track.
+ *
+ * @param {Object} state - The state of the app.
+ * @returns {Object}
+ */
+export function getVideoTrack(state: Object): Object {
+    return state['features/prejoin'].videoTrack;
+}
+
+/**
+ * Selector for getting the mute status of the prejoin audio.
+ *
+ * @param {Object} state - The state of the app.
+ * @returns {boolean}
+ */
+export function isPrejoinAudioMuted(state: Object): boolean {
+    return state['features/prejoin'].audioMuted;
+}
+
+/**
+ * Selector for getting the name that the user filled while configuring.
+ *
+ * @param {Object} state - The state of the app.
+ * @returns {boolean}
+ */
+export function getPrejoinName(state: Object): string {
+    return state['features/prejoin'].name;
+}
+
+/**
+ * Selector for getting the mute status of the prejoin video.
+ *
+ * @param {Object} state - The state of the app.
+ * @returns {boolean}
+ */
+export function isPrejoinVideoMuted(state: Object): boolean {
+    return state['features/prejoin'].videoMuted;
+}
+
+/**
+ * Selector for getting the error if any while creating streams.
+ *
+ * @param {Object} state - The state of the app.
+ * @returns {string}
+ */
+export function getRawError(state: Object): string {
+    return state['features/prejoin'].rawError;
+}
+
+/**
+ * Selector for getting state of the prejoin audio.
+ *
+ * @param {Object} state - The state of the app.
+ * @returns {boolean}
+ */
+export function isAudioDisabled(state: Object): Object {
+    return state['features/prejoin'].audioDisabled;
+}
+
+/**
+ * Selector for getting state of the prejoin video.
+ *
+ * @param {Object} state - The state of the app.
+ * @returns {boolean}
+ */
+export function isPrejoinVideoDisabled(state: Object): Object {
+    return state['features/prejoin'].videoDisabled;
+}
+
+/**
+ * Selector for getting the visiblity state for the 'JoinByPhoneDialog'.
+ *
+ * @param {Object} state - The state of the app.
+ * @returns {boolean}
+ */
+export function isJoinByPhoneDialogVisible(state: Object): boolean {
+    return state['features/prejoin'].showJoinByPhoneDialog;
+}
+
+/**
+ * Returns true if the prejoin page is enabled and no flag
+ * to bypass showing the page is present.
+ *
+ * @param {Object} state - The state of the app.
+ * @returns {boolean}
+ */
+export function isPrejoinPageEnabled(state: Object): boolean {
+    return (
+        state['features/base/config'].prejoinPageEnabled
```

Comment:
None of these 2 options are present in config.js and the whitelist. They must be.

Reply:
I have added it to the config & whitelist.